### PR TITLE
side navigation changes for notifications

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
* This PR adds side navigation changes to notification plugin when feature is enabled

### Issues Resolved

After side nav changes, when feature is enabled

Image 7-18-24 at 5 39 PM (1)
Image 7-18-24 at 5 39 PM

When feature is disabled:

Image 7-18-24 at 7 22 AM

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.ts [hunk 1]
```diff
@@ -61,6 +71,117 @@ export class notificationsDashboardsPlugin
       });
     }
 
+    if (core.chrome.navGroup.getNavGroupEnabled()) {
+
+      // register applications with category and use case information
+      core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.observability, [
```
Comment (SuZhou-Joe) [inline public/plugin.ts:77]: Notification features should be only put into settings and setup nag group.

Comment (riysaxen-amzn) [reply public/plugin.ts:77]: @amsiglan  ^^

Comment (riysaxen-amzn) [reply public/plugin.ts]: addressed

**Review by AWSHurneyt — APPROVED** [review]

File: public/plugin.ts [hunk 2]
```diff
@@ -61,6 +71,66 @@ export class notificationsDashboardsPlugin
...
+        id: `email_groups`,
+        title: 'Email recepient groups',
+        order: 9090,
+        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
+        mount: async (params: AppMountParameters) => {
+          return mountWrapper(params, ROUTES.EMAIL_GROUPS);
+        },
+      });
+
+      const navLinks = [
```
Comment (amsiglan) [inline public/plugin.ts:114]: Nit: Shorten this to 
```
const navlinks = ['', '', ''].map(id => ({ id, parent: PLUGIN }))

File: public/plugin.ts [hunk 3]
```diff
@@ -61,6 +71,66 @@ export class notificationsDashboardsPlugin
...
+        mount: async (params: AppMountParameters) => {
+          return mountWrapper(params, ROUTES.EMAIL_SENDERS);
+        },
+      });
+
+      core.application.register({
+        id: `email_groups`,
+        title: 'Email recepient groups',
+        order: 9090,
+        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
```
Comment (amsiglan) [inline public/plugin.ts:108]: What is workspaceAvailability? and what are the other options used for? Could this be `workspace/no-workspace` distinction?

File: public/plugin.ts [hunk 4]
```diff
@@ -33,19 +35,27 @@ export class notificationsDashboardsPlugin
...
       order: 9060,
+      workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
       async mount(params: AppMountParameters) {
         // Load application bundle
         const { renderApp } = await import('./application');
         // Get start services as specified in opensearch_dashboards.json
         const [coreStart, depsStart] = await core.getStartServices();
         // Render the application
-        return renderApp(coreStart, params, dataSourceManagement, depsStart);
+        return renderApp(coreStart, params, dataSourceManagement!, depsStart, ROUTES.CHANNELS);
```
Comment (amsiglan) [inline public/plugin.ts:58]: Should this be alerts?

**Review by SuZhou-Joe — APPROVED** [review]

File: public/plugin.ts [hunk 5]
```diff
@@ -61,6 +71,63 @@ export class notificationsDashboardsPlugin
       });
     }
 
+    if (core.chrome.navGroup.getNavGroupEnabled()) {
+      // register applications with category and use case information
+      core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.settingsAndSetup, [
+        {
+          id: PLUGIN_NAME,
+          showInAllNavGroup: true
```
Comment (SuZhou-Joe) [inline public/plugin.ts:79]:
```suggestion
```
notifications channels is not expected to show in all nav group.
